design-document.md

```diff
@@ -106,6 +123,167 @@ to achieve these goals:
+1. If clients reveal identifying information to aggregators (such as a trusted
+   identity during client authentication), aggregators can learn which clients
+   are contributing data.
+     1. Aggregators may weaken anonymity by revealing that a particular client
+        contributed data to the system.
+     1. Aggregators may choose to selectively omit data from certain clients.
+          * For example, omitting submissions from a particular geographic
+            region to falsely suggest that a particular localization is not
+            being used.
+1. Individual aggregators may compromise availability of the system by refusing
+   to emit aggregation parts.
+
+##### Mitigations
+
+1. The linear secret sharing scheme employed by the client ensures that privacy
+   is preserved unless every participating aggregator reveals their unencrypted
+   data shares.
+1. Infrastructure diversity. Prio deployments should ensure that aggregators do
+   not have common dependencies that would enable a single vendor to reassemble
+   data.
+     * For example, if all participating aggregators stored unencrypted data
+       shares on the same cloud object storage service, then that cloud vendor
+       would be able to reassemble all the data shares and defeat privacy.
+1. Running the protocol over multiple subsets of the available aggregators
+   chosen so that no aggregator appears in all subsets and accepting any result
+   would prevent an individual aggregator from compromising availability.
+1. Running the protocol over multiple subsets of the available aggregators
+   chosen so that no aggregator appears in all subsets and requiring that the
+   results agree would prevent an individual aggregator from compromising
+   robustness
+     * This could also allow identifying of defective aggregators, by finding
+       the aggregator that appears in every subset whose results were wrong.
+
+#### Leader
+
+The leader is also an aggregator, and so all the assets, capabilities and
+mitigations available to aggregators also apply to the leader.
+
+##### Capabilities
+
+1. Data validity proof verification. The leader can forge proofs and collude
+   with a malicious client to trick aggregators into aggregating invalid data.
+1. Relaying messages between aggregators. The leader can compromise availability
+   by dropping messages.
+
+##### Mitigations
+
+1. Rather than having all aggregators trust the leader to validate proofs, each
+   aggregator could exchange proof shares with every other aggregator, enabling
+   each aggregator to independently verify the proofs.
+1. Running the protocol over multiple subsets of the available aggregators and
+   choosing different leaders for each run would prevent any individual leader
+   from compromising availability.
+
+#### Collector
+
+##### Capabilities
+
+1. Advertising shared configuration parameters (e.g., minimum thresholds for
+   aggregations, polynomial identity test values, arithmetic circuits).
```

Comment:
In [BBC+19] (and hence libprio-rs) the concept of "polynomial identity test" is generalized to distributed evaluation of a PCP over joint randomness. Hence, I would use the term "joint randomness" instead of "polynomial identity test values".

To your point: We haven't yet decided how the joint randomness is chosen. It could very well be the collector, but this is TBD. (See "Consensus protocol" below.) Also, I would add "designation the leader role" to this list.
But